src/metabase/xrays/automagic_dashboards/interesting.clj

```diff
@@ -100,43 +100,49 @@
   ->reference (fn [template-type model]
                 [template-type (mi/model model)]))
 
-(defn- optimal-datetime-resolution
+(defn- optimal-temporal-resolution
   [field]
   (let [[earliest latest] (some->> field
                                    :fingerprint
                                    :type
                                    :type/DateTime
                                    ((juxt :earliest :latest))
-                                   (map u.date/parse))]
+                                   (map u.date/parse))
+        date?             (isa? (:base_type field) :type/Date)
+        date-time?        (isa? (:base_type field) :type/DateTime)
+        time?             (isa? (:base_type field) :type/Time)]
     (if (and earliest latest)
-      ;; e.g. if 3 hours > [duration between earliest and latest] then use `:minute` resolution
-      (condp u.date/greater-than-period-duration? (u.date/period-duration earliest latest)
-        (t/hours 3) :minute
-        (t/days 7) :hour
-        (t/months 6) :day
-        (t/years 10) :month
-        :year)
-      :day)))
+      (let [duration      (u.date/period-duration earliest latest)
+            greater-than? #(u.date/greater-than-period-duration? % duration)]
```

Comment:
Since we're comparing "in reverse", I would find it more natural to call this `less-than?`

Reply:
yes 21e7926103